packages/jest-cli/src/jest.js

```diff
@@ -228,8 +228,7 @@ const runJest = (config, argv, pipe, testWatcher, onComplete) => {
       .then(runResults => {
         if (config.testResultsProcessor) {
           /* $FlowFixMe */
-          const processor = require(config.testResultsProcessor);
-          processor(runResults);
+          runResults = require(config.testResultsProcessor)(runResults);
```

Comment:
I don't really want to be backwards compatible though. I want to pay down the technical debt. I still feel guilty for originally merging this PR :(